opm/autodiff/ParallelOverlappingILU0.hpp

```diff
@@ -65,10 +69,91 @@ struct ConstructionTraits<Opm::ParallelOverlappingILU0<Matrix,Domain,Range,Paral
 
 } // end namespace Amg
 
+
+
 } // end namespace Dune
 
 namespace Opm
 {
+    namespace detail
+    {
+      //! compute ILU decomposition of A. A is overwritten by its decomposition
+      template<class M, class CRS, class InvVector>
+      void convertToCRS(const M& A, CRS& lower, CRS& upper, InvVector& inv )
+      {
+        typedef typename M :: size_type size_type;
+
+        // count non-zeros in case 0 is returned
+        // this is due to a bug in dune-istl < 2.5
+        size_type nonZeros = A.nonzeroes();
+        if( nonZeros == 0 )
+        {
+            // count non-zeros
+            const auto end = A.end();
+            for( auto row = A.begin(); row != end; ++row )
+            {
+                nonZeros += (*row).size();
+            }
+        }
+
+        lower.resize( A.N(), nonZeros );
+        upper.resize( A.N(), nonZeros );
```

Comment:
The bilu_decomposition could be done directly using the new storage. That would be the best. 
Your suggested factor should work but I'll implement an reserve as you go strategy. That is usually the best and does not require shrinkToFit later.